### ISSUE TITLE
Validation Set Distributed Sampling when using WebDataset with FSDP

### ISSUE THREAD

**Issue by adamcatto** [webdataset/webdataset#364]
I have a collection of tar files with validation-set samples (I have calculated and saved the number of validation samples to a variable `num_val_samples`) that I load in with a webdataset DataPipeline as follows:

```py
val_dataset = wds.DataPipeline(
        wds.ResampledShards(
            os.path.join('path', 'to',  'val_samples_{0000...xxxx}.tar')
        ),
        wds.tarfile_to_samples(),
        wds.decode(),
        # at this point, we have an list of decompressed training samples from each shard in this worker in sequence
        wds.to_tuple("input.npy", "target.npy"),
        wds.batched(1)
    ).with_length(num_val_samples)
```

I also have a multi-node setup and am using FSDP to train the model. I essentially want each GPU to process `num_val_samples / num_gpus`, but my understanding is that WebDataset doesn't play nicely with PyTorch DistributedSampler.

In my case, I actually have an equal number of shards (tar files) as GPUs, although one is about half the size of the others. Would it make most sense to distribute each shard to a separate GPU, or is there a nice way that I can distribute samples to equalize the work across GPUs despite some shards having less samples?
